linera-views/tests/operations_tests.rs

```diff
@@ -205,103 +208,180 @@ async fn test_readings_memory_specific() {
+                let delete_key_prefix = get_small_key_space(rng, key_prefix, len);
                 batch.delete_key_prefix(delete_key_prefix.clone());
+            }
+        }
+    }
+    if option == 2 {
+        // A renormalized batch.
+        let batch = generate_random_batch(rng, key_prefix, size_batch, 1);
+        let unord_batch = batch.simplify();
+        return unord_batch.as_batch();
+    }
+    Batch::new()
+}
+
+fn get_key(key_prefix: &[u8], key_suffix: Vec<u8>) -> Vec<u8> {
+    let mut key = key_prefix.to_vec();
+    key.extend(key_suffix);
+    key
+}
+
+fn generate_specific_batch(key_prefix: &[u8], option: usize) -> Batch {
+    let mut batch = Batch::new();
+    if option == 1 {
+        let key = get_key(key_prefix, vec![34]);
+        batch.put_key_value_bytes(key.clone(), vec![]);
+        batch.delete_key(key);
+    }
+    if option == 2 {
+        let key1 = get_key(key_prefix, vec![12, 34]);
+        let key2 = get_key(key_prefix, vec![12, 33]);
+        let key3 = get_key(key_prefix, vec![13]);
+        batch.put_key_value_bytes(key1.clone(), vec![]);
+        batch.put_key_value_bytes(key2, vec![]);
+        batch.put_key_value_bytes(key3, vec![]);
+        //
+        batch.delete_key(key1);
+        let key_prefix = get_key(key_prefix, vec![12]);
+        batch.delete_key_prefix(key_prefix);
+    }
+    batch
+}
+
+fn realize_batch(batch: &Batch) -> BTreeMap<Vec<u8>, Vec<u8>> {
+    let mut kv_state = BTreeMap::new();
+    for operation in &batch.operations {
+        match operation {
+            WriteOperation::Put { key, value } => {
+                kv_state.insert(key.to_vec(), value.to_vec());
+            }
+            WriteOperation::Delete { key } => {
+                kv_state.remove(key);
+            }
+            WriteOperation::DeletePrefix { key_prefix } => {
                 let key_list = kv_state
-                    .range(get_interval(delete_key_prefix.clone()))
+                    .range(get_interval(key_prefix.clone()))
                     .map(|x| x.0.to_vec())
                     .collect::<Vec<_>>();
                 for key in key_list {
                     kv_state.remove(&key);
                 }
             }
         }
-        key_value_store.write_batch(batch, &[]).await.unwrap();
-        // Checking the consistency
-        let mut key_values = BTreeMap::new();
-        for key_value in key_value_store
-            .find_key_values_by_prefix(&key_prefix)
-            .await
-            .unwrap()
-            .iterator()
-        {
-            let key_value = key_value.unwrap();
-            let mut key = key_prefix.clone();
-            key.extend(key_value.0);
-            key_values.insert(key, key_value.1.to_vec());
-        }
-        assert_eq!(key_values, kv_state);
+    }
+    kv_state
+}
+
+#[cfg(test)]
+async fn run_test_batch<OP: KeyValueStoreClient + Sync>(
+    key_value_operation: &OP,
```

Comment:
```suggestion
    kv_client: &C,
```

Reply:
This was a residue from previous work. In a previous PR, I renamed it as `key_value_store` and  think that is a fine name.